o2checkcode.sh

```diff
@@ -19,26 +19,33 @@ cp "${O2_ROOT}"/compile_commands.json .
 O2_SRC=$(python -c 'import json,sys,os; sys.stdout.write( os.path.commonprefix([ x["file"] for x in json.loads(open("compile_commands.json").read()) if not "G__" in x["file"] and x["file"].endswith(".cxx") ]) )')
 [[ -e "$O2_SRC"/CMakeLists.txt && -d "$O2_SRC"/.git ]]
 
-BASECOMMIT=${O2CHECKCODE_BASECOMMIT:=${ALIBUILD_PR_BASE}}
-HEADCOMMIT=${O2CHECKCODE_HEADCOMMIT:=${ALIBUILD_PR_HEAD}}
+# We have something to compare our working directory to. We check only the
+# changed files (including the untracked ones) if the list of relevant files
+# that changed is up to 50 entries long. User can trigger a check by setting the
+# upstream reference hash in the O2CHECKCODE_UPSTREAM_HASH. User can also export
+# O2CHECKCODE_WORK_HASH if she does not want to compare the current head plus
+# the working directory.
+ALIBUILD_UPSTREAM_HASH=${ALIBUILD_UPSTREAM_HASH:=$O2CHECKCODE_UPSTREAM_HASH}
```

Comment:
I find `UPSTREAM` still a bit confusing. (But I don't want to block this any longer either).